File: <100/036/036.cpp
#include<iostream>
#include<algorithm>
using namespace std;


string to_binary(int n){
  string s = "";
  while(n > 0){
    s += to_string(n&1);
    n >>= 1;
  }

  return s;
}


bool is_palindrome(string s){
  string t = s;
  reverse(t.begin(), t.end());
  return t == s;
}


bool both_palindrome(int n){
  return is_palindrome(to_string(n)) && is_palindrome(to_binary(n));
}


int main(){
  int answer = 0;
  const int upper_limit = 1e6;
  for(int n = 1; n < upper_limit; ++n){
    if(both_palindrome(n)) answer += n;
  }

  cout << answer << endl;
  return 0;
}

File: <100/036/036.py
import time


def to_str_bin(n: int) -> str:
    return f'{n:b}'


def is_palindrome(n: int, format: callable) -> bool:
    s = list(format(n))
    t = list(format(n))
    t.reverse()
    return s == t


def both_palindrome(n: int) -> bool:
    return is_palindrome(n, str) and is_palindrome(n, to_str_bin)


def main() -> int:
    upper_limit = 10**6
    answer = sum(n for n in range(upper_limit) if both_palindrome(n))
    return answer


if __name__ == "__main__":
    start = time.time()
    answer = main()
    elapsed = time.time() - start

    print(f'{elapsed} sec')
    print(answer)
